### PR TITLE
Feature/sign in

### PR DESCRIPTION
**Pull request by aguskoll**
### What's in

The sign in with the backend and in the app.
Error messages when the sign in fails.
Unit testing
View binding

### How it looks

https://drive.google.com/file/d/1ZCzOPVzA2flNn5IJbpeHvJBQWduRECZN/view?usp=sharing

### How to test

1. Run this branch
2. Create an account 
3. Go to the sign in page and fill up your data
4. Verify that you can sign in correctly.
5. Sign out and fill your data incorrectly
6. Verify the error message

### CODE REVIEW

File: app/src/main/java/com/rootstrap/android/ui/activity/main/SignInActivity.kt [hunk 1]
```diff
@@ -47,46 +52,54 @@ class SignInActivity : PermissionActivity(), AuthView {
...
-                // TODO..
-            }
+            override fun denied() = Unit
 
-            override fun foreverDenied() {
-                // TODO..
-            }
+            override fun foreverDenied() = Unit
         })
     }
```
Comment (amaury901130) [inline app/src/main/java/com/rootstrap/android/ui/activity/main/SignInActivity.kt]: this is an example function about how to ask permissions using PermissionActivity, removed it if not needed.

File: app/src/main/java/com/rootstrap/android/ui/activity/main/SignInActivityViewModel.kt [hunk 2]
```diff
@@ -13,30 +13,40 @@ import com.rootstrap.android.util.ViewModelListener
...
+) : BaseViewModel(listener) {
 
     var state: SignInState = SignInState.none
         set(value) {
             field = value
             listener?.updateState()
         }
 
+    fun canSignIn(userName: String?, password: String?): Boolean {
+        return userName.isNullOrEmpty().not() && password.isNullOrEmpty().not()
```
Comment (amaury901130) [inline app/src/main/java/com/rootstrap/android/ui/activity/main/SignInActivityViewModel.kt]: add password validation: password length should be more than 6 characters.

**Review by amaury901130 — APPROVED** [review]
💪 great, just a couple of suggestions.

**Review by ximenaperez — APPROVED** [review]

File: app/src/main/java/com/rootstrap/android/ui/activity/main/SignInActivityViewModel.kt (hunk 2 above)
Comment (aguskoll) [reply app/src/main/java/com/rootstrap/android/ui/activity/main/SignInActivityViewModel.kt]: done!

File: app/src/main/java/com/rootstrap/android/ui/activity/main/SignInActivity.kt (hunk 1 above)
Comment (aguskoll) [reply app/src/main/java/com/rootstrap/android/ui/activity/main/SignInActivity.kt]: done!

**Pull request merged**